proposed/2021/NuGetConsumePDBInPackageReference.md

```diff
@@ -0,0 +1,216 @@
+Currently, NuGet does not select pdb or xml files.
+
+In comparison, in packages.config, .pdb, .xml files are automatically copied from lib folder into the build output folder because of the [ResolveAssemblyReference](https://github.com/dotnet/msbuild/blob/main/src/Tasks/AssemblyDependency/ResolveAssemblyReference.cs#L75) task. So there is no such problem in packages.config.
+
+## Who are the customers
+
+Partner team(.NET SDK) who needs to consume .pdb, .xml files for their features in PackageReference world. 
+
+Take debugging as an example. When a NuGet package includes a .pdb file in its lib folder, in order to debug into the NuGet package, it needs the .pdb file to be copied to the build output folder of the project which references the NuGet package.
+
+## Requirements
+
+* A feature to allow the .NET SDK to consume .pdb, .xml files.
+
+## Goals
+
+* Define an approach to make .NET SDK be able to consume .pdb, .xml files.
+
+## Non-Goals
+
+* .NET SDK side change to consume .pdb or .xml.
+
+## Solution
+
+For any given assembly under lib and runtime folder from a package reference, if there are files next to it that differ only by extension, NuGet will add a "related" property underneath the assembly in targets section of assets file, listing the extensions of these files, seperated by `;`.
+
+* Applies to all the compile time assemblies and runtime assemblies.
+* Apply to transitive dependencies.
+* Only apply to package reference. Will not apply to project reference.
+
+
+### Solution - Technical details
+
+#### Package reference:
+For example, one project has a package reference of PackageA. Suppose in folder ./PackageA/1.0.0/lib/netstandard2.0, there are PackageA.pdb, PackageA.xml file besides PackageA.dll.
+Currently, the targets section in assets file is:
+
+```json
+{
+  "version": 3,
+  "targets": {
+    ".NETCoreApp,Version=v5.0": {
+      "PackageA/1.0.0": {
+        "type": "package",
+        "compile": {
+           "lib/netstandard2.0/PackageA.dll": {}
+        },
+        "runtime": {
+           "lib/netstandard2.0/PackageA.dll": {}
+      }
+    }
+  },
+```
+When this solution is applied, the targets section in assets file will be:
+
+```json
+{
+  "version": 3,
+  "targets": {
+    ".NETCoreApp,Version=v5.0": {
+      "PackageA/1.0.0": {
+        "type": "package",
+        "compile": {
+           "lib/netstandard2.0/PackageA.dll": {
+              "related": "pdb;xml"
+           }
+        },
+        "runtime": {
+           "lib/netstandard2.0/PackageA.dll": {
+              "related": "pdb;xml"
+           }
+      }
+    }
+  },
+```
+#### Project reference:
+For example, one project has a project reference of ProjectA. Supppose in folder ./ProjectA/lib/net5.0, there are PackageA.pdb, PackageA.xml file besides ProjectA.dll.
+There will be no change in targets section in assets file for project reference:
```

Comment:
```suggestion
There will be no change in the assets file for project references:
```

Reply:
Can you also explain _why_? Is it just because RAR will already do the right thing for ProjectReference?

Reply:
Thanks for the suggestion!
Yes, the ResolveAssemblyReference task will copy the related files, without adding any new info into assets file.
I added the explanation in the doc.